### PR TITLE
Address Dockerfile Security Recommendations

### PR DESCRIPTION
**Pull request by mansinahar**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -25,12 +25,14 @@ RUN go build -mod=vendor -ldflags "-X github.com/prebid/prebid-server/version.Ve
...
-COPY static static/
-COPY stored_requests/data stored_requests/data
+COPY --chmod=a+xr --from=build /app/prebid-server .
+COPY --chmod=a+r static static/
+COPY --chmod=a+r stored_requests/data stored_requests/data
 RUN apt-get update && \
     apt-get install -y ca-certificates mtr && \
     apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+RUN adduser prebid_user
+USER prebid_user
```
Comment (SyntaxNode) [inline Dockerfile:35]: Does `prebid_user` need any explicit permissions?

**Review by SyntaxNode — COMMENTED** [review]
Thank you for hardening the Docker images, @mansinahar. Did you test this by creating and running the image locally?

File: Dockerfile [hunk 2]
```diff
@@ -25,12 +25,14 @@ RUN go build -mod=vendor -ldflags "-X github.com/prebid/prebid-server/version.Ve
...
-COPY --from=build /app/prebid-server/prebid-server .
-COPY static static/
-COPY stored_requests/data stored_requests/data
+COPY --chmod=a+xr --from=build /app/prebid-server .
+COPY --chmod=a+r static static/
+COPY --chmod=a+r stored_requests/data stored_requests/data
 RUN apt-get update && \
     apt-get install -y ca-certificates mtr && \
     apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+RUN adduser prebid_user
```
Comment (mansinahar) [inline Dockerfile:34]: Executing prebid-server with a non-root user

File: Dockerfile [hunk 3]
```diff
@@ -25,12 +25,14 @@ RUN go build -mod=vendor -ldflags "-X github.com/prebid/prebid-server/version.Ve
 FROM ubuntu:18.04 AS release
 LABEL maintainer="hans.hjort@xandr.com" 
 WORKDIR /usr/local/bin/
-COPY --from=build /app/prebid-server/prebid-server .
-COPY static static/
-COPY stored_requests/data stored_requests/data
+COPY --chmod=a+xr --from=build /app/prebid-server .
```
Comment (mansinahar) [inline Dockerfile:28]: Making Prebid Server binary and other static resources to be non-writable

**Comment by SyntaxNode** [comment]
Looks good. Please make the same changes to `.devcontainer/Dockerfile`.

**Comment by mansinahar** [comment]
> Looks good. Please make the same changes to `.devcontainer/Dockerfile`.

I tried this but I am getting an issue when executing the apt-get commands with a non-root user as it requires permission to sybdirectories under /var/lib. This is the exact error:
```
> [4/4] RUN apt-get update && export DEBIAN_FRONTEND=noninteractive     && apt-get -y install --no-install-re
[2022-02-09T15:44:30.767Z] commends vim:
#7 0.253 Reading package lists...
#7 0.265 E: List directory /var/lib/apt/lists/partial is missing. - Acquire (13: Permission denied)
```
This also got me thinking that since this is a dev container, meant only for development purposes, executing with a root user should be okay, no? I am not sure what other issues will be uncovered by moving it to be executed by a non-root user. I think it's good to have the flexibility for development purposes. Thoughts @SyntaxNode ?

**Comment by SyntaxNode** [comment]
> Thoughts @SyntaxNode ?

Ah, ok. Makes sense.

**Review by SyntaxNode — APPROVED** [review]

**Review by bsardo — APPROVED** [review]
LGTM

File: Dockerfile (hunk 3 above)
Comment (mansinahar) [reply Dockerfile:28]: Looks like this was released as part of 20.10 but was somehow missed from release notes (See: https://github.com/moby/moby/issues/34819#issuecomment-749570216) and requires `DOCKER_BUILDKIT=1` for it to work correctly (See: https://github.com/moby/moby/issues/34819#issuecomment-765637751).

We will change this to use the `RUN` command for now as this is a relatively new feature and might break for a lot of other people. Thanks for calling it out!

**Pull request merged**